battle.go

```diff
@@ -108,7 +108,22 @@ func (b *Battle) Start() error {
 
 // Handles all pre-turn logic
 func (b *Battle) preRound() {
-	// TODO
+	for a, party := range b.parties {
+		for ap, pkmn := range party.activePokemon {
+			t := target{
+				party:     a,
+				partySlot: ap,
+				Pokemon:   *pkmn,
+				Team:      party.team,
+			}
+			if v, ok := pkmn.metadata[MetaSleepTime]; ok && v.(int) == 0 && pkmn.StatusEffects.check(StatusSleep) {
+				b.QueueTransaction(CureStatusTransaction{
+					Target:       t,
+					StatusEffect: StatusSleep,
+				})
+			}
+		}
+	}
```

Comment:
This particular case can be simplified to `for _, t := range b.GetTargets()` but I understand if we want to keep this for future cases.